sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -360,24 +369,35 @@ private[sql] class ParquetRelation2(
       commonMetadataStatuses =
         leaves.filter(_.getPath.getName == ParquetFileWriter.PARQUET_COMMON_METADATA_FILE)
 
-      // If we already get the schema, don't need to re-compute it since the schema merging is
-      // time-consuming.
-      if (dataSchema == null) {
```

Comment:
This is irrelevant to this PR, but I'd like to point out that this check is unnecessary now. IIRC, at the time PR #6012 was merged, `ParquetRelation2` still needed to be serialized to executor side, thus avoiding schema merging on executor side makes sense. However, after migrating to `HadoopFsRelation`, `ParquetRelation2` won't be serialized anymore (you may see that `ParquetRelation2` is no longer a case class and doesn't extend `Serializable` now).